src/plugins/esql_datagrid/public/data_grid.tsx

```diff
@@ -0,0 +1,143 @@
+
+interface ESQLDataGridProps {
+  core: CoreStart;
+  data: DataPublicPluginStart;
+  fieldFormats: FieldFormatsStart;
+  rows: ESQLRow[];
+  dataView: DataView;
+  columns: DatatableColumn[];
+  query: AggregateQuery;
+  flyoutType?: 'overlay' | 'push';
+  isTableView?: boolean;
+}
+type DataTableColumnsMeta = Record<
+  string,
+  {
+    type: DatatableColumnMeta['type'];
+    esType?: DatatableColumnMeta['esType'];
+  }
+>;
+
+const DataGrid: React.FC<ESQLDataGridProps> = (props) => {
+  const storage = new Storage(localStorage);
+  const [expandedDoc, setExpandedDoc] = useState<DataTableRecord | undefined>(undefined);
+  const [activeColumns, setActiveColumns] = useState<string[]>(
+    props.isTableView ? props.columns.map((c) => c.name) : []
+  );
+  const [rowHeight, setRowHeight] = useState<number>(5);
+
+  if (props.dataView.fields.getByName('@timestamp')?.type === 'date') {
+    props.dataView.timeFieldName = '@timestamp';
+  }
+
+  const renderDocumentView = useCallback(
+    (
+      hit: DataTableRecord,
+      displayedRows: DataTableRecord[],
+      displayedColumns: string[],
+      customColumnsMeta?: DataTableColumnsMeta
+    ) => (
+      <RowViewer
+        dataView={props.dataView}
+        toastNotifications={props.core.notifications}
+        hit={hit}
+        hits={displayedRows}
+        columns={displayedColumns}
+        columnsMeta={customColumnsMeta}
+        flyoutType={props.flyoutType ?? 'push'}
+        onRemoveColumn={(column) => {
+          setActiveColumns(activeColumns.filter((c) => c !== column));
+        }}
+        onAddColumn={(column) => {
+          setActiveColumns([...activeColumns, column]);
+        }}
+        onClose={() => setExpandedDoc(undefined)}
+        setExpandedDoc={setExpandedDoc}
+      />
+    ),
+    [activeColumns, props.core.notifications, props.dataView, props.flyoutType]
+  );
+
+  const columnsMeta = props.columns.reduce((acc, column) => {
+    acc[column.id] = {
+      type: column.meta?.type,
+      esType: column.meta?.esType ?? column.meta?.type,
+    };
+    return acc;
+  }, {} as DataTableColumnsMeta);
+
+  const columnNames = props.columns?.map(({ name }) => name);
+  const rows: DatatableRow[] = props.rows.map((row) => zipObject(columnNames, row));
+
+  return (
+    <UnifiedDataTable
+      columns={activeColumns}
+      rows={rows.map((row: Record<string, string>, idx: number) => {
+        return {
+          id: String(idx),
+          raw: row,
+          flattened: row,
+        } as unknown as DataTableRecord;
+      })}
+      columnsMeta={columnsMeta}
+      services={{
+        data: props.data,
+        theme: props.core.theme,
+        uiSettings: props.core.uiSettings,
+        toastNotifications: props.core.notifications.toasts,
+        fieldFormats: props.fieldFormats,
+        storage,
+      }}
+      isPlainRecord
+      isSortEnabled={false}
+      loadingState={DataLoadingState.loaded}
+      dataView={props.dataView}
+      sampleSizeState={500}
```

Comment:
Is it really 500?

Reply:
Good catch, I did some experiments and forgot it